src/cmap/connect.ts

```diff
@@ -119,93 +119,92 @@ function performInitialHandshake(
 
   const authContext = new AuthContext(conn, credentials, options);
   conn.authContext = authContext;
-  prepareHandshakeDocument(authContext, (err, handshakeDoc) => {
-    if (err || !handshakeDoc) {
-      return callback(err);
-    }
-
-    const handshakeOptions: Document = Object.assign({}, options);
-    if (typeof options.connectTimeoutMS === 'number') {
-      // The handshake technically is a monitoring check, so its socket timeout should be connectTimeoutMS
-      handshakeOptions.socketTimeoutMS = options.connectTimeoutMS;
-    }
-
-    const start = new Date().getTime();
-    conn.command(ns('admin.$cmd'), handshakeDoc, handshakeOptions, (err, response) => {
-      if (err) {
-        callback(err);
-        return;
+  prepareHandshakeDocument(authContext).then(
+    handshakeDoc => {
+      const handshakeOptions: Document = Object.assign({}, options);
+      if (typeof options.connectTimeoutMS === 'number') {
+        // The handshake technically is a monitoring check, so its socket timeout should be connectTimeoutMS
+        handshakeOptions.socketTimeoutMS = options.connectTimeoutMS;
       }
 
-      if (response?.ok === 0) {
-        callback(new MongoServerError(response));
-        return;
-      }
+      const start = new Date().getTime();
+      conn.command(ns('admin.$cmd'), handshakeDoc, handshakeOptions, (err, response) => {
```

Comment:
Should we use commandAsync here?

Reply:
doesn't this function take a callback? there's no reason to use the async version unless we're in a method that's explicitly async